helpers.sh

```diff
@@ -94,7 +94,7 @@ function set_base_lmp_version {
 	# 3: Find our base LMP version based on the HEAD
 	export LMP_VERSION=$(git describe --tags --abbrev=0 HEAD)
 	export LMP_VERSION_CACHE="$LMP_VERSION"
-	if [[ "${H_PROJECT}" == "lmp" ]] ; then
+	if [[ "${H_PROJECT}" == "lmp" ]] || [ -v LMP_VERSION_CACHE_DEV ] ; then
```

Comment:
Can you explain how LMP_VERSION_CACHE_DEV would be used in this context?

I guess this is how you would have a factory using the dev cache?

Reply:
right, is to use the dev cache in the factory
```
The variable LMP_VERSION_CACHE_DEV is to use the development version of the cache:

# factory-config.yml

  params:
    IMAGE: lmp-factory-image
    SSTATE_CACHE_MIRROR: ""
    LMP_VERSION_CACHE_DEV: ""
```
I have already be using this for some time in some of my factories, the `lmp-ci-testing-next` that uses the latest next as well as on the `reproducible` that uses the latest stable.

Reply:
The `setup-environment-internal` can handle the `LMP_VERSION_CACHE_DEV`  but only when the `LMP_VERSION_CACHE` is not defined so for the ci build I add to add it too